ios-xcframework/build.sh

```diff
@@ -18,13 +24,105 @@ function _xcodebuild {
     fi
 }
 
-DERIVED_DATA_PATH=./DerivedData
-DESTINATION='generic/platform=iOS'
+function log {
+  if [[ $# -eq 1 ]]; then
+    PREFIX=''
+    MESSAGE=$1
+  else
+    PREFIX=":$1: "
+    MESSAGE=$2
+  fi
```

Comment:
I think it's best to declare the variables within a function body (the ones above and the `PLATFORM` in the `archive` function) as `local` to prevent them from being leaked outside of this function.

Here is a test script:
```shell
#!/bin/bash

function use_global_var {
    GLOBAL_VAR="global var is visible"
}

function use_local_var {
    local local_var; local_var="local var is invisible"
}

use_global_var
use_local_var

echo ${GLOBAL_VAR:-nope} # Output: global var is visible
echo ${local_var:-nope} # Output: nope
```

Reply:
TIL, thanks!

Reply:
I'll merge this as is, and followup with a dedicated PR.